rust/src/delta.rs

```diff
@@ -1316,4 +1328,17 @@ mod tests {
         assert_eq!(2, *state.app_transaction_version.get("abc").unwrap());
         assert_eq!(1, *state.app_transaction_version.get("xyz").unwrap());
     }
+
+    macro_rules! aw {
+        ($e:expr) => {
+            tokio_test::block_on($e)
+        };
+      }
+
+    #[test]
+    fn opens_table_with_history() {
+        let mut table = aw!(open_table("tests/data/delta-0.8.0")).unwrap();
```

Comment:
you should be able to write an async test like this: https://github.com/delta-io/delta-rs/blob/main/rust/tests/read_delta_test.rs#L9-L13